overlord/state/taskrunner.go

```diff
@@ -146,6 +146,16 @@ func (r *TaskRunner) KnownTaskKinds() []string {
 	return kinds
 }
 
+// TaskKindHasUndo returns true if the indicated kind has an undo handler.
+func (r *TaskRunner) TaskKindHasUndo(k string) bool {
+	for kind, h := range r.handlers {
+		if kind == k {
+			return h.undo != nil
+		}
+	}
+	return false
+}
```

Comment:
If I'm reading this correctly, `r.handlers` is a `map[string]handlePair` which maps from kind to handler. So shouldn't we be able to look up the kind directly?

```suggestion
func (r *TaskRunner) TaskKindHasUndo(k string) bool {
	if handler, ok := r.handlers[k]; ok {
		return h.undo != nil
	}
	return false
}
```